src/main/java/ttt/board/LineGenerator.java

```diff
@@ -4,46 +4,92 @@
 
 public class LineGenerator {
     private final PlayerSymbol[] grid;
+    private int dimension;
 
     public LineGenerator(PlayerSymbol... grid) {
         this.grid = grid;
+        this.dimension = (int) Math.sqrt(grid.length);
     }
 
     public Line[] linesForAllDirections() {
-        return new Line[] {topRow(), middleRow(), bottomRow(),
-                leftColumn(), middleColumn(), rightColumn(),
-                backslashDiagonal(), forwardslashDiagonal()};
-    }
+        Line[] allLines = new Line[(dimension * 2) + 2];
```

Comment:
Is there any value in using an array here? Using a `List<Line>` would save you the noisy sizing of the array.